### PR TITLE
Remove `Forwardable` dependency (possible conversion to bundled gem)

### PR DESCRIPTION
**Pull request by MSP-Greg**
### Description

[Forwardable](https://github.com/ruby/forwardable) is currently used by server.rb & client.rb.

It is part of the standard library, but it's also available as a gem. Hence, it could be 'converted' to a bundled gem, especially since it's nor an extension gem.

Bundled gems need to be included as dependencies in gemspec files.

Since removing it is relatively trivial (5 methods are delegated), lets do so now.

Closes #3190
Closes https://github.com/puma/puma/issues/3218

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [ ] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: lib/puma/client.rb [hunk 1]
```diff
@@ -110,7 +108,10 @@ def initialize(io, env=nil)
 
     attr_accessor :remote_addr_header, :listener
 
-    def_delegators :@io, :closed?
+    # Remove in Puma 7?
+    def closed?
+      @to_io.closed?
```
Comment (dentarg) [inline lib/puma/client.rb:113]: Why the change from `@io` to `@to_io`?

(Maybe that's better, I don't know)

**Comment by dentarg** [comment]
I think it would be good if the commit message for this included the motivation behind doing it

**Comment by dentarg** [comment]
It isn't really spelled out in #3190 but there is this 

> It is part of the standard library, but it's also available as a gem. Hence, it could be 'converted' to a bundled gem. Bundled gems need to be included as dependencies in gemspec files.

And we want to avoid the use of bundled gems in Puma so phased restart works well, right?

**Comment by MSP-Greg** [comment]
> And we want to avoid the use of bundled gems in Puma so phased restart works well, right?

Yes, but that's really more critical with extension gems, which I don't think can be reloaded.  If a bundled gem is a dependency, we'd need to include it in Puma's gemspec for Bundler to properly handle it.

I can't recall what gems/repos it involved, but `racc` was recently converted to a bundled gem, and it broke CI here and there...

I'll add some of the comments from the discussion to the PR.  Thanks.

File: lib/puma/client.rb (hunk 1 above)
Comment (MSP-Greg) [reply lib/puma/client.rb:113]: Sorry, I composed a reply, but must have closed the window.

Ruby's TCP & UNIX sockets inherit from `IO`, and `IO#to_io` returns `self`.  Puma's MiniSSL sockets do not inherit from `IO`, just like Ruby's OpenSSL sockets.

Their `closed?` method checks the underlying TCP socket.  So, rather than add another Ruby method call, I used `@to_io` to get the TCP socket.

Note that both `io` and `to_io` are read-only attributes of `Client`, so these can be removed without anyone losing functionality.  As mentioned, that would still be an API change.

**Review by nateberkopec — APPROVED** [review]
LGTM assuming that the commit message will be changed as dentarg suggested
